Signal/src/contact/SystemContactsFetcher.swift

```diff
@@ -373,32 +373,40 @@ class SystemContactsFetcher: NSObject {
      *
      * @param   completion  completion handler is called on main thread.
      */
-    public func requestOnce(completion: ((Error?) -> Void)?) {
+    public func requestOnce(completion completionParam: ((Error?) -> Void)?) {
         AssertIsOnMainThread()
 
+        // Ensure completion is invoked on main thread.
+        let completion = { error in
```

Comment:
We weren't always ensuring that the completion was invoked on main.

Reply:
This also avoids the `?()`

Reply:
> We weren't always ensuring that the completion was invoked on main.
  
Huh, I can't see which case we missed. It looks like they're all invoked on main. 

But I guess with this approach at worst we dispatch an empty block which is NBD.

Reply:
The missed cases are in `updateContacts`, but I decided to rework the completion handling in both methods with the same goal: ensure the that completion is always called on main thread and DRY.